v2/tools/generator/internal/astmodel/conversion_function_builder.go

```diff
@@ -397,10 +424,37 @@ func IdentityConvertComplexMapProperty(
 		Body:  astbuilder.StatementBlock(conversion...),
 	}
 
-	result := astbuilder.IfNotNil(
-		params.GetSource(),
-		makeMapStatement,
-		rangeStatement)
+	// If we must forcibly construct empty collections, check if the destination is nil and if so, construct an empty collection
+	// This only applies for top-level collections (we don't forcibly construct nested collections)
+	var result *dst.IfStmt
+	if depth == 0 && builder.ForceEmptyCollections {
+		emptyMap := astbuilder.MakeMap(keyTypeAst, valueTypeAst)
+
+		helpfulComment := &dst.EmptyStmt{
+			Decs: dst.EmptyStmtDecorations{
+				NodeDecs: dst.NodeDecs{
+					End: []string{fmt.Sprintf("// Set property to empty map, as this resource is set to serialize all collections explicitly")},
+				},
+			},
+		}
```

Comment:
This doesn't need to be separate like this; instead, add the comment directly to the `SimpleAssignment()` you have below. That's also where you can ensure there's a line break prior to the comment (as I noted elsewhere).

Other cases need the same change.

Reply:
Fixed